src/public/modules/forms/helpers/CsvMergedHeadersGenerator.ts

```diff
@@ -108,17 +131,20 @@ class CsvMergedHeadersGenerator extends CsvGenerator {
     // Craft a new csv row for each unprocessed record
     // O(qn), where q = number of unique questions, n = number of submissions.
     this.unprocessed.forEach((up) => {
-      let createdAt = moment(up.created).tz('Asia/Singapore')
-      createdAt = createdAt.isValid()
+      const createdAt = moment(up.created).tz('Asia/Singapore')
+      const formattedDate = createdAt.isValid()
         ? createdAt.format('DD MMM YYYY hh:mm:ss A')
-        : createdAt
-      let row = [up.submissionId, createdAt]
-      for (let [fieldId] of this.fieldIdToQuestion) {
-        const numCols = this.fieldIdToNumCols[fieldId]
-        for (let colIndex = 0; colIndex < numCols; colIndex++) {
-          row.push(this._extractAnswer(up.record, fieldId, colIndex))
-        }
-      }
+        : createdAt.toString() // just convert to string if given date is not valid
+      const row = [up.submissionId, formattedDate]
+
+      this.fieldIdToQuestion.forEach(
+        (_question: { created: string; question: string }, fieldId: string) => {
```

Comment:
you don't actually need to specify these types. since `fieldIdToQuestion` is a typed Map, its `forEach` is also automatically typed.